framework/src/application/node/forger/forger.js

```diff
@@ -220,7 +233,56 @@ class Forger {
 					)} is not a delegate`,
 				);
 			}
+			// Prepare hash-onion
+			const registeredHashOnion = registeredHashOnions[account.address];
+			const { hashOnion: hashOnionConfig } =
+				this.config.forging.delegates.find(
+					d => getAddressFromPublicKey(d.publicKey) === account.address,
+				) || {};
+			if (!hashOnionConfig) {
+				throw new Error(
+					`Account ${account.address} does not have hash onion in the config`,
+				);
+			}
+			const configHashOnionSeed =
+				hashOnionConfig.hashes[hashOnionConfig.hashes.length - 1];
+			if (registeredHashOnion && registeredHashOnion !== configHashOnionSeed) {
+				this.logger.warn(
+					`Hash onion for Account ${account.address} is not the same as previous one. Overwriting with new hash onion`,
+				);
+				usedHashOnions = usedHashOnions.filter(
+					ho => ho.address !== account.address,
+				);
+			}
+			registeredHashOnions[account.address] = configHashOnionSeed;
```

Comment:
When overriding hash onion for a delegate, does the above line not come inside if?

Reply:
In case nothing is registered, it should also register the seed